### PR TITLE
Replace @rnw-scripts/jest-out-of-tree-resolver with @rnx-kit/jest-resolver

### PR DESCRIPTION
**Pull request by tido64**
###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8314)

### CODE REVIEW

File: change/@rnw-scripts-promote-release-e733b348-2393-45e5-a8a3-5b374988df71.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (NickGerleman) [inline change/@rnw-scripts-promote-release-e733b348-2393-45e5-a8a3-5b374988df71.json:1]: It looks like these got applied to some packages they shouldn't. This sometimes happens when origin points to the non-msft fork, and beachball looks in the wrong place. You might need to pass something like "--branch upstream/main" when doing a yarn change.

File: change/@rnw-scripts-promote-release-e733b348-2393-45e5-a8a3-5b374988df71.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  "type": "none",
+  "comment": "Set executable bit so the script can be executed on non-Windows file systems",
```
Comment (NickGerleman) [inline change/@rnw-scripts-promote-release-e733b348-2393-45e5-a8a3-5b374988df71.json:3]: nit: consider leaving a more meaningful change description

**Comment by NickGerleman** [comment]
There are some files modified that look like they are zero line changes. Wondering if line endings were modified somehow?

**Comment by tido64** [comment]
> There are some files modified that look like they are zero line changes. Wondering if line endings were modified somehow?

The file mode was changed from 644 to 755. This happens if you check out RNW and run `yarn` on a *nix machine. It will set the executable bit on all scripts. I'm checking in the changes otherwise I would have to reset them every time I run `yarn`. It's just the meta data of the files that changed. The content is untouched.

**Comment by NickGerleman** [comment]
> > There are some files modified that look like they are zero line changes. Wondering if line endings were modified somehow?
> 
> The file mode was changed from 644 to 755. This happens if you check out RNW and run `yarn` on a *nix machine. It will set the executable bit on all scripts. I'm checking in the changes otherwise I would have to reset them every time I run `yarn`. It's just the meta data of the files that changed. The content is untouched.

Sounds reasonable

**Comment by dannyvv** [comment]
@NickGerleman : Consideration for our backlog: Run yarn install on a unix machine and check if `git add` wants to add something?

**Review by NickGerleman — APPROVED** [review]

**Pull request merged**